### PR TITLE
image uri fix

### PR DESCRIPTION
**Pull request by elpie89**
This should be ok
I also changed the UI as requested
#685

### CODE REVIEW

**Comment by Drigax** [comment]
Thanks for the fix, I'll take a look at this first thing tomorrow!

**Comment by Drigax** [comment]
Alright, looks good for the most part, I'd just like to add a warning to our output window when we are placing textures in a directory that's not in a subdirectory of the glTF

File: 3ds Max/Max2Babylon/Forms/ExporterForm.cs [hunk 1]
```diff
@@ -138,25 +138,30 @@ private void butModelBrowse_Click(object sender, EventArgs e)
 
             if (saveFileDialog.ShowDialog(this) == DialogResult.OK)
             {
-                txtModelName.MaxPath(saveFileDialog.FileName);
+                txtModelPath.MaxPath(saveFileDialog.FileName);
+                if (string.IsNullOrWhiteSpace(txtTexturesPath.Text))
+                {
+                    string defaultTexturesDir = Path.GetDirectoryName(txtModelPath.Text);
+                    txtTexturesPath.MaxPath(defaultTexturesDir);
+                }
```
Comment (Drigax) [inline 3ds Max/Max2Babylon/Forms/ExporterForm.cs]: Can we add a second check here along the lines of:
```

                if (!PathUtilities.IsBelowPath(txtTexturesPath.Text, txtModelPath.Text))
                {
                    {
                        CreateWarningMessage("WARNING: textures path should be below model file path, not all client renderer support this feature",0);
                    }
                }
```

File: 3ds Max/Max2Babylon/Forms/ExporterForm.cs [hunk 2]
```diff
@@ -175,14 +180,14 @@ private void btnTextureBrowse_Click(object sender, EventArgs e)
             if (dialog.ShowDialog() == CommonFileDialogResult.Ok)
             {
                 string selectedFolderPath = dialog.FileName;
-                string absoluteModelPath = txtModelName.Text;
+                string absoluteModelPath = txtModelPath.Text;
 
                 if (!PathUtilities.IsBelowPath(selectedFolderPath, absoluteModelPath))
                 {
-                    MessageBox.Show("WARNING: folderPath should be below model file path");
+                    CreateWarningMessage("WARNING: textures path should be below model file path, not all client renderer support this feature",0);
```
Comment (Drigax) [inline 3ds Max/Max2Babylon/Forms/ExporterForm.cs]: NIT: can you change "client renderer" to "client renderers"?

File: SharedProjects/Babylon2GLTF/GLTFExporter.Texture.cs [hunk 3]
```diff
@@ -160,7 +160,16 @@ private GLTFTextureInfo ExportTexture(BabylonTexture babylonTexture, GLTF gltf,
...
-                        textureUri = PathUtilities.GetRelativePath( exportParameters.outputPath,exportParameters.textureFolder) + "/"+ name;
+                        textureUri = PathUtilities.GetRelativePath( exportParameters.outputPath,exportParameters.textureFolder);
+                        if (textureUri == ".\\")
+                        {
+                            //texture is in same location of model
+                            textureUri = name;
+                        }
+                        else
+                        {
+                            textureUri = Path.Combine(textureUri, name);
```
Comment (Drigax) [inline SharedProjects/Babylon2GLTF/GLTFExporter.Texture.cs]: On windows, we end up having backslashes in our image URI path, after getting this relative path. should we replace that with forward slashes?

File: SharedProjects/Babylon2GLTF/GLTFExporter.cs [hunk 4]
```diff
@@ -482,7 +482,7 @@ private List<GLTFBufferView> SwitchImagesFromUriToBinary(GLTF gltf)
 
             foreach (GLTFImage gltfImage in gltf.ImagesList)
             {
-                var path = Path.Combine(gltf.OutputFolder, Uri.UnescapeDataString(gltfImage.uri));
+                var path = Path.Combine(gltf.OutputFolder, gltfImage.uri);
```
Comment (Drigax) [inline SharedProjects/Babylon2GLTF/GLTFExporter.cs]: Why are we no longer escaping the symbols in our path?

Comment (elpie89) [reply SharedProjects/Babylon2GLTF/GLTFExporter.cs]: it is already unescaped in PathGetRelativePath..this was a repetition.
the gltfimage uri is set only in L176

**Comment by elpie89** [comment]
Everything is now managed in GetRelativePath() method
the relative part is calculated using uring and I removed the part where forward slashes were transformed in backslashes
the Path.Combine() does not alter the slash..so everything is now forwarded.
I think we should stay with URI format, as windows accept both.

Also, I do not remove the first "./"
It is useless and the URI format accept it

the second warning has been added too, in case a model path is set after the textures one

we end up with this 3 cases:

```
"uri": "../../texture/checker.jpg" (with warning raised)
"uri": "./checker.jpg"
"uri": "texture/checker.jpg"
```

If I did not forget some case, should be ok

File: SharedProjects/Babylon2GLTF/GLTFExporter.cs [hunk 5]
```diff
@@ -470,7 +470,11 @@ private string gltfToJson(GLTF gltf)
             // Use the bounded writer in case some values are infinity ()
             using (var jsonWriter = new JsonTextWriterBounded(sw))
             {
+#if DEBUG
```
Comment (Drigax) [inline SharedProjects/Babylon2GLTF/GLTFExporter.cs]: Great change!

**Pull request merged**